### PR TITLE
updated `integrations/providers/google`

### PR DESCRIPTION
**Pull request by leo-gan**
Added missed integrations. Updated formats.

### CODE REVIEW

File: docs/docs/integrations/platforms/google.mdx [hunk 1]
```diff
@@ -170,33 +248,91 @@ documents = docai_wh_retriever.get_relevant_documents(
...
+from langchain.tools import GooglePlacesTool
+```
+
 ### Google Search
 
-- Install requirements with `pip install google-api-python-client`
+We need to install a python package.
+
+```bash
+pip install google-api-python-clien
```
Comment (baskaryan) [inline docs/docs/integrations/platforms/google.mdx:286]: client

File: docs/docs/integrations/platforms/google.mdx [hunk 2]
```diff
@@ -170,33 +248,91 @@ documents = docai_wh_retriever.get_relevant_documents(
...
 We can easily load this wrapper as a Tool (to use with an Agent). We can do this with:
+
 ```python
 from langchain.agents import load_tools
 tools = load_tools(["google-search"])
 ```
 
-## Document Transformer
-### Google Document AI
+### Google Serper
```
Comment (baskaryan) [inline docs/docs/integrations/platforms/google.mdx:304]: think this is https://serper.dev (uses google under the hood but not developed by google)

**Comment by holtskinner** [comment]
I have more updates/changes to be made in https://github.com/langchain-ai/langchain/pull/12224

File: docs/docs/integrations/platforms/google.mdx (hunk 2 above)
Comment (holtskinner) [reply docs/docs/integrations/platforms/google.mdx:304]: Correct - This shouldn't be included

Comment (leo-gan) [reply docs/docs/integrations/platforms/google.mdx:304]: "All functionality related to Google Cloud Platform and other `Google` products."

Users don't care if the access is implemented by the Google library or by a third-party library; it is an "internal" detail. They just want to access Google Search. 
For example, the same as with `UnstructuredWordDocumentLoader`, `unstructured` is not a Microsoft product but a layer between a user and Microsoft Word.

Comment (baskaryan) [reply docs/docs/integrations/platforms/google.mdx:304]: i think in the providers/platforms section specifically folks are looking for tools created by those providers/platforms. if we do want to keep stuff like this here then i'd strongly prefer a separate `3rd Party Integrations` section at the bottom, it's important for folks to know this isn't provided by Google.

**Comment by leo-gan** [comment]
@baskaryan I've implemented your comments

**Review by efriis — APPROVED** [review]

**Pull request merged**